### PR TITLE
[vcpkg_common_definitions] Add macro VCPKG_HOST_IS_<target>

### PR DESCRIPTION
**Pull request by JackBoosY**
Provide host architecture macros to the cross-platform compilation environment instead of CMAKE_HOST_WIN32 etc.
- VCPKG_HOST_IS_WINDOWS
- VCPKG_HOST_IS_OSX
- VCPKG_HOST_IS_LINUX
- VCPKG_HOST_IS_DARWIN

Related: #10913.

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]
LGTM

**Comment by JackBoosY** [comment]
blocked by #13765.

**Review by BillyONeal — APPROVED** [review]

File: scripts/cmake/vcpkg_common_definitions.cmake [hunk 1]
```diff
@@ -45,6 +46,17 @@ elseif(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "MinGW")
...
 
+#Helper variables to identify the host system name
+if (CMAKE_HOST_WIN32)
+    set(VCPKG_HOST_IS_WINDOWS 1)
+elseif (CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
+    set(VCPKG_HOST_IS_OSX 1)
+elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+    set(VCPKG_HOST_IS_LINUX 1)
+elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "FreeBSD")
+    set(VCPKG_HOST_IS_FreeBSD 1)
```
Comment (dan-shaw) [inline scripts/cmake/vcpkg_common_definitions.cmake:57]: Should we rename this to `VCPKG_HOST_IS_FREEBSD`?

**Review by dan-shaw — APPROVED** [review]

File: scripts/cmake/vcpkg_common_definitions.cmake (hunk 1 above)
Comment (JackBoosY) [inline scripts/cmake/vcpkg_common_definitions.cmake:57]:
```suggestion
    set(VCPKG_HOST_IS_FREEBSD 1)
```

**Pull request merged**